Modules/posixmodule.c

```diff
@@ -4530,6 +4530,86 @@ os_listmounts_impl(PyObject *module, path_t *volume)
 }
 
 
+/*[clinic input]
+os._path_isdevdrive
+
+    path: path_t
+
+Determines whether the specified path is on a Windows Dev Drive.
+
+[clinic start generated code]*/
+
+static PyObject *
+os__path_isdevdrive_impl(PyObject *module, path_t *path)
+/*[clinic end generated code: output=1f437ea6677433a2 input=ee83e4996a48e23d]*/
+{
+#ifndef PERSISTENT_VOLUME_STATE_DEV_VOLUME
+    const int PERSISTENT_VOLUME_STATE_DEV_VOLUME = 0x00002000;
```

Comment:
Can you add a comment hyperlink to an MSDN or similar thing where this constant is defined if not in the local headers?

Reply:
Also, is it guaranteed to be a CPP macro? Because it seems to me this would fail if MS decides to implement it as a regular global instead.

Reply:
> Can you add a comment hyperlink to an MSDN or similar thing where this constant is defined if not in the local headers?

Will do, though the docs are... lagging. It'll be in the headers in the associated Windows SDK update, but this way we don't need to require contributors to update all the way just to build.

> Also, is it guaranteed to be a CPP macro?

Yes, it's a regular macro. I've seen the updated header already ;-)

Reply:
The [`FILE_FS_PERSISTENT_VOLUME_INFORMATION`](https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/ntifs/ns-ntifs-_file_fs_persistent_volume_information) struct and its associated `VolumeFlags` macros are documented online for use by drivers in kernel mode. In the WDK, the struct and flags are defined in "ntifs.h". In the SDK, they're defined in "winioctl.h" for use with WinAPI `DeviceIoControl()`. The online documentation sometimes takes a long time to catch up with additions to the WDK/SDK. Currently it only documents up to `PERSISTENT_VOLUME_STATE_BACKED_BY_WIM` (0x00000040), but the SDK defines an additional 6 flags (0x00000080 to 0x00001000).

Support for Dev Drive is available only in preview builds of the next release of Windows 11. The current SDK for Windows 11 22H2 (10.0.22621) thus does not define `PERSISTENT_VOLUME_STATE_DEV_VOLUME` (0x00002000).